Reference/Routing/custom-controllers.md

```diff
@@ -2,167 +2,246 @@
+
+In this example, the Product Page Document Type has two templates 'ProductPage' and 'ProductAmpPage', we can hijack and handle the requests to the two templates differently.
+
+Creating the Controller as before
 
 ```csharp
-public class HomeController : Umbraco.Web.Mvc.RenderMvcController
+public class ProductPageController : Umbraco.Web.Mvc.RenderMvcController
 {
-    public ActionResult MobileHomePage(ContentModel model)
+    // Any request for the 'ProductAmpPage' template will be handled by this Action
+    public ActionResult ProductAmpPage(ContentModel model)
+    {
+        // Create AMP specific content here...
+        return CurrentTemplate(model);
+    }
+    // All other request, eg the ProductPage template will be handled by the default 'Index' action
+    public override ActionResult Index(ContentModel model)
     {
-        // Do some stuff here, the return the base Index method
-        return base.Index(model);
+        // you are in control here!
+
+        // return a 'model' to the selected template/view for this page.
+        return CurrentTemplate(model);
     }
 }
 ```
+#### How can a page be requested via two different templates?
+The page in Umbraco will have a single 'template' selected as it's default template, but it's possible to call this same page on a different template by adding ?altTemplate=othertemplatename to the Url QueryString eg:
+
+/products/superfancyproduct/?altTemplate=ProductAmpPage
+
+would request the Super Fancy Product page but force it to be loaded with the ProductAmpPage template, this can be simplified further:
+
+/products/superfancyproduct/ProductAmpPage/
 
-## How the mapping works
+where the final part of the url is assumed to be the altTemplate querystring value.
+
+### Summary - How the route hijacking convention works
 
 * Document Type name = controller name
 * Template name = action name (if no action matches or is not specified - then the 'Index' action will be executed).
+* Controller Inherits from `Umbraco.Web.Mvc.RenderMvcController`
 
 ## Returning a view with a custom model
 
-If you want to return a custom model to a view then there are a few steps that need to be taken.
+The steps to achieve this will differ, depending if your template views are using IPublishedContent or Modelsbuilder generated Models.
 
 ### Changing the @inherits directive of your template
 
-First, the standard view that is created by Umbraco inherits from `Umbraco.Web.Mvc.UmbracoViewPage` which has a model defined of type `Umbraco.Web.Models.ContentModel`. You'll see the inherits directive at the top of the view as:
+By default, your Umbraco Template will be based on the `ContentModel` that the default `RenderMvcController` passes through to it. (NB: ContentModel has one property called 'Content' of type IPublishedContent, and the @Model syntax in V8 is wired up to use this IPublishedContent item, to save having to type @Model.Content.etc as in V7)
```

Comment:
The parentheses here - could that by any chance be converted to a note instead? 🤔 Just an idea.

Reply:
done